### DISCUSSION TITLE
How to use ASL on Conductor?

### DISCUSSION THREAD

**Discussion by taoj-action** [Netflix/conductor#2215] [Q&A]
Hi, We are adopting Netflix Conductor as a workflow engine but we have quite a lot of work based on ASL (Amazon State Language). Since Conductor runs its decider on the server-side, Is translating ASL to Conductor DSL on demand the only solution? Thanks!

**Comment by aravindanr** [comment]
Conductor does not directly support Amazon State Language. If you could elaborate your use-case, we can be of help.

**Reply by taoj-action** [reply]
We have a large investment in modeling in ASL and in-house workflow engine (not distributed one). We are not using Step Function. Now, we are moving to Conductor but its DSL is different than ASL. One possibility is to write a converter to convert to Conductor DSL when doing any CRUD operation on ASL workflow. However, it doesn't seem to be a straghtforward task. Is there any better option available?

I also asked a related question on CNCF’s serverless workflow for reference. Thanks!
https://github.com/serverlessworkflow/specification/issues/358

**Reply by aravindanr** [reply]
@taojwmware There is no converter available from ASL to Conductor's DSL. While writing the converter, if you have questions on Conductor DSL, we are here to help.
